forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/rs/api/payment/v3_1_4/domesticpayments/DomesticPaymentConsentsApiController.java

```diff
@@ -22,17 +22,136 @@
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.forgerock.openbanking.aspsp.rs.wrappper.RSEndpointWrapperService;
+import com.forgerock.openbanking.common.model.openbanking.persistence.payment.FRDomesticConsent;
 import com.forgerock.openbanking.common.services.store.RsStoreGateway;
 import com.forgerock.openbanking.common.services.store.payment.DomesticPaymentService;
+import com.forgerock.openbanking.exceptions.OBErrorResponseException;
+import org.joda.time.DateTime;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.server.ServletServerHttpRequest;
 import org.springframework.stereotype.Controller;
+import uk.org.openbanking.datamodel.payment.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.Principal;
+import java.util.Collections;
+
+import static com.forgerock.openbanking.common.utils.ApiVersionUtils.getOBVersion;
+import static uk.org.openbanking.datamodel.service.converter.payment.OBWriteDomesticConsentConverter.toOBWriteDomesticConsent2;
 
 @Controller("DomesticPaymentConsentsApiV3.1.4")
-public class DomesticPaymentConsentsApiController extends com.forgerock.openbanking.aspsp.rs.api.payment.v3_1_3.domesticpayments.DomesticPaymentConsentsApiController implements DomesticPaymentConsentsApi {
+public class DomesticPaymentConsentsApiController implements DomesticPaymentConsentsApi {
+
+    private final RSEndpointWrapperService rsEndpointWrapperService;
+    private final RsStoreGateway rsStoreGateway;
+    private final DomesticPaymentService paymentsService;
+    private final ObjectMapper mapper;
 
     public DomesticPaymentConsentsApiController(RSEndpointWrapperService rsEndpointWrapperService,
                                                 RsStoreGateway rsStoreGateway,
                                                 DomesticPaymentService paymentsService,
-                                                ObjectMapper objectMapper) {
-        super(rsEndpointWrapperService, rsStoreGateway, paymentsService, objectMapper);
+                                                ObjectMapper mapper) {
+        this.rsEndpointWrapperService = rsEndpointWrapperService;
+        this.rsStoreGateway = rsStoreGateway;
+        this.paymentsService = paymentsService;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticConsentResponse4> createDomesticPaymentConsents(
+            OBWriteDomesticConsent4 obWriteDomesticConsent4,
```

Comment:
I'm trying to recall if I missed how this changed from `OBWriteDomesticConsent3`, or if I deliberately ignored the ReadRefundAccount enum (or if I generated the model from an earlier version of the 3.1.4 spec?!). I remember there was no mention of it in the announcement (aside from a reverse payments proposition, which I'm not sure is the same thing?): https://www.openbanking.org.uk/about-us/latest-news/open-banking-publishes-the-latest-version-of-the-open-banking-standard-2019/

I'll see if I can spot any other changes, but let me know if you've seen any.